tensorflow/core/kernels/conv_ops_3d.cc

```diff
@@ -495,21 +511,22 @@ struct LaunchConvOp<GPUDevice, T> {
       algorithm_config.set_scratch_size(best_result.scratch_size());
 #endif
       AutoTuneConv3d::GetInstance()->Insert(conv_parameters, algorithm_config);
-    }
+    } else {
```

Comment:
so introducing this else block essentially prevents `ThenConvolveWithAlgorithm` from being called twice in the first go around (i.e. when we have a autotune cache miss).

What is the rationale for doing this? this change will get flagged when we try to upstream it, and we need to have a good reason for doing so.

Reply:
I feel the same that it is going to cause trouble when up-streaming the code.

On the other hand, something I realize is that calling `ThenConvolveWithAlgorithm()` twice can be a quite expensive operation. In the specific 3d convolution config I've tested with, `DoPrepareForConvolution()` will allocate 5GB of gpu memory for each call. With `ThenConvolveWithAlgorithm()` calling twice, it blows up TensorFlow. 

I do realize that the config I've tested with is rare, and for regular configs, it only cause trouble for the first miss, so I'm neutral to either keep/remove it.